sync-core/src/test/java/com/cloudant/sync/replication/PullReplicatorTest.java

```diff
@@ -69,4 +72,53 @@ public void start_StartedThenComplete() throws InterruptedException {
         Assert.assertTrue(listener.finishCalled);
         Assert.assertFalse(listener.errorCalled);
     }
+
+    @Test
+    public void testRequestFilters() throws Exception {
+
+        //to test the filters we make an invalid request
+
+        TestReplicationListener listener = new TestReplicationListener();
+        PullReplication pullReplication = createPullReplication();
+        pullReplication.addRequestFilters(new InvalidJSONFilter());
+        Replicator replicator = ReplicatorFactory.oneway(pullReplication);
+        replicator.getEventBus().register(listener);
+        replicator.start();
+
+        while(replicator.getState() != Replicator.State.COMPLETE && replicator.getState() != Replicator.State.ERROR) {
+            Thread.sleep(1000);
+        }
+
+        Assert.assertEquals(Replicator.State.ERROR, replicator.getState());
+        Assert.assertFalse(listener.finishCalled);
+        Assert.assertTrue(listener.errorCalled);
+        Assert.assertTrue(listener.exception instanceof CouchException);
+        CouchException couchException = (CouchException) listener.exception;
+        Assert.assertTrue(couchException.getStatusCode() == 400);
+        Assert.assertEquals(couchException.getReason(), "invalid_json");
+
+    }
+
+    @Test
+    public void testResponseFilters() throws Exception {
+        //to test the filters we make an invalid request
```

Comment:
comment doesn't seem to match code - it's a response filter so how can it make an invalid request?